model/vm_host.rb

```diff
@@ -67,6 +67,30 @@ def ip6_random_vm_network(prefix = 79)
     end
   end
 
+  def ip4_random_vm_network
+    subnet = NetAddr::CIDR.create(ipv4_bucket)
+    ip_list = subnet.enumerate
+  
+    # remove the first and the last IP from the range
+    ip_list.shift
+    ip_list.pop
+  
+    # split the remaining IPs into subnets of 4 IPs each
+    subnets = ip_list.each_slice(4).map do |slice|
```

Comment:
```suggestion
    subnets = ip_list.each_slice(4).filter_map do |slice|
```
removes need to call `.compact` at end